src/main/webapp/populateData.js

```diff
@@ -143,17 +143,36 @@ function populateCalendar() {
       })
       .then((hoursJson) => {
         // Display the days and the free hours for each one of them
-        const days = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'];
-        for (let day = 0; day < 5; day++) {
-          const dayContainer = document.querySelector('#day-'+day);
-          dayContainer.innerText = days[(hoursJson.startDay + day) % 7];
-          const workContainer = document.querySelector('#work-day-'+day);
-          workContainer.innerText =
-              convertTime(hoursJson.workHoursPerDay[day]) + ' (working)';
-          const personalContainer =
-              document.querySelector('#personal-day-'+day);
-          personalContainer.innerText =
-              convertTime(hoursJson.personalHoursPerDay[day]) + ' (personal)';
+        const days = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'];
+        const panelContent = document.querySelector('#panel-content');
+        panelContent.innerHTML = '';
+        for (const day in hoursJson.workTimeFree) {
+          if (typeof day == 'string') {
+            const panelContentEntry = document.createElement('div');
+            panelContentEntry.className = 'panel__content-entry';
+            const dayContainer = document.createElement('p');
+            dayContainer.className = 'panel__text-icon u-text-calendar';
+            dayContainer.innerText =
+                days[(hoursJson.startDay + parseInt(day)) % 7];
+            const timeContainer = document.createElement('div');
+            const workContainer = document.createElement('p');
+            workContainer.className = 'u-block';
+            workContainer.innerText =
+                hoursJson.workTimeFree[day].hours +
+                'h ' + hoursJson.workTimeFree[day].minutes +
+                'm free (working)';
+            const personalContainer = document.createElement('p');
+            personalContainer.className = 'u-block';
+            personalContainer.innerText =
+                hoursJson.personalTimeFree[day].hours +
+                'h ' + hoursJson.personalTimeFree[day].minutes +
+                'm free (personal)';
+            timeContainer.appendChild(workContainer);
+            timeContainer.appendChild(personalContainer);
+            panelContentEntry.appendChild(dayContainer);
+            panelContentEntry.appendChild(timeContainer);
+            panelContent.appendChild(panelContentEntry);
```

Comment:
It may be simpler to just use a template literal[1] and write out the desired HTML in its source form, then append it to the `panelContent.innerHTML`. However, this approach is technically correct - it's just hard to understand.

You are welcome to submit as-is and file an issue to clean this up later.

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
I will file an issue to clean this up later.